### PR TITLE
Add debian support

### LINKED ISSUE #47: Add Debian support

### PR DESCRIPTION
**Pull request by TheDoctorsLife**
Closes #47
This needs to be tested, everything should work just fine however.

### CODE REVIEW

**Comment by TheDoctorsLife** [comment]
This is ready to be merged.

File: migrations/20160511124741_add_void_support/down.sql [hunk 1]
```diff
@@ -3,10 +3,11 @@ CREATE TABLE new_packages (
   arch TEXT NOT NULL,
   aur TEXT NOT NULL,
   ubuntu TEXT NOT NULL,
-  ubuntu_dev TEXT NOT NULL
+  ubuntu_dev TEXT NOT NULL,
+  debian TEXT NOT NULL,
```
Comment (mgattozzi) [inline migrations/20160511124741_add_void_support/down.sql]: You don't need to change old migrations. the new one adds it. the date in order is how diesel figures out how to change the db. It's sort of like a branch of commits.

File: migrations/20160514215928_add_debian_support/up.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+ALTER TABLE packages ADD COLUMN debian TEXT NOT NULL DEFAULT ''; UPDATE packages SET debian = 'sudo' WHERE id = '1' --Keep repeating the UPDATE statment until all packages are added
```
Comment (mgattozzi) [inline migrations/20160514215928_add_debian_support/up.sql]: Style wise can you break this into two lines?

File: scripts/debian_setup.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+#Install nightly
+rustup install nightly
+
+#Set Directory as nightly
+rustup override add nightly-2016-05-08
+
+#Install diesel_cli and db
+cargo install diesel_cli
+~/.cargo/bin/diesel setup
+echo "Add .cargo/bin to your PATH if you want to use diesel_cli everywhere"
```
Comment (mgattozzi) [inline scripts/debian_setup.sh]: This is fine for now. I'll make sure to remove this last bit once I get the build script setup to sue this

File: src/alchemy/debian.rs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+//Apitude specific functions
+
+/// Calls the aptitude program to install packages
+///
+/// #Examples
+///
+/// ```
+/// let mut packages: HashSet<String> = HashSet::new();
+/// packages.push("sudo".to_owned());
```
Comment (mgattozzi) [inline src/alchemy/debian.rs]: This is my bad. I need to change this to be insert elsewhere in the docs. You probably just added it in.

File: src/alchemy/distro.rs [hunk 5]
```diff
@@ -59,6 +61,12 @@ pub fn which_distro() -> Option<Distro> {
     if arch.is_ok() && arch.unwrap().is_file() {
         return Some(Distro::Arch)
     }
+    //Check for Debian differently but only after the
+    //previous method trys by using /etc/arch-release
+    let debian = fs::metadata("/etc/debian_version");
+    if debian.is_ok() && debian.unwrap().is_file() {
```
Comment (mgattozzi) [inline src/alchemy/distro.rs]: Is debian_version a file on the system?

**Comment by mgattozzi** [comment]
This looks pretty much done. Make sure the migration works and go through my fixes and you should be okay to merge. Just post that it works in this pr and you can do it yourself.

File: migrations/20160511124741_add_void_support/down.sql (hunk 1 above)
Comment (TheDoctorsLife) [inline migrations/20160511124741_add_void_support/down.sql]: Thats the down.sh.

File: migrations/20160514215928_add_debian_support/up.sql (hunk 2 above)
Comment (TheDoctorsLife) [inline migrations/20160514215928_add_debian_support/up.sql]: Sure.

File: scripts/debian_setup.sh (hunk 3 above)
Comment (TheDoctorsLife) [inline scripts/debian_setup.sh]: I don't have all the dependices yet for debian, besides that, your right the script is fine.

File: src/alchemy/distro.rs (hunk 5 above)
Comment (TheDoctorsLife) [inline src/alchemy/distro.rs]: Yes.

File: migrations/20160511124741_add_void_support/down.sql (hunk 1 above)
Comment (mgattozzi) [inline migrations/20160511124741_add_void_support/down.sql]: down.sh? This is the down.sql file

**Pull request merged**